File: src/PTUI.Core/Entities/Setting.cs
using System.ComponentModel.DataAnnotations;
using PTUI.Core.Enums;
using PTUI.Core.Model;

namespace PTUI.Core.Entities;

public class Setting
{
    [Required]
    [Key]
    public Guid Id { get; set; }
    [Required]
    public string Name { get; set; }
    //public string DefaultValue { get; set; }
    public string Definition { get; set; }
    public SettingType Type { get; set; }
    public IList<SettingValue> Values { get; set; }

    public Setting(string name, string definition, SettingType type)
    {
        Id = new Guid();
        Name = name;
        Definition = definition;
        Type = type;
    }
}

File: src/PTUI.WebApp/src/interfaces/Setting.ts
import {ISettingValue} from "./SettingValue";

export interface ISetting {
    id : string;
    name: string;
    definition: string;
    type: number;
    values: ISettingValue[];
}

export class Setting implements ISetting {
    id : string;
    name: string;
    definition: string;
    type: number;
    values: ISettingValue[];

    constructor(id: string, name: string, definition: string, type: string, values: ISettingValue[]) {
        this.id = id;
        this.name = name;
        this.definition = definition;
        this.type = Number(type);
        this.values = values;
    }
}